### PR TITLE
Remove Guava test library

### PR DESCRIPTION
**Pull request by basil**
`guava-testlib` naturally depends on Guava. Putting it on our test classpath just increases our dependency on Guava and complicates upgrading Guava or doing fancy things like shading or removing Guava. We really don't need this library. It was just used in one test to facilitate `.equals()` testing. I rewrote that test to do the same thing without Guava.

I'm not sure why this library was ever exposed in the core BOM, because it's a test-scoped dependency. I am removing it from there as well. There are only two plugins that use this library in their tests: [Gravatar](https://github.com/jenkinsci/gravatar-plugin) and [Octoperf](https://github.com/jenkinsci/octoperf-plugin). Each one declare its dependency on this library in its own `pom.xml` file, so they should be unaffected by removal of this library from the core BOM.

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Comment by TobiX** [comment]
There is also the quite excellent [EqualsVerifier](https://jqno.nl/equalsverifier/) if you want to avoid boilerplate `equals` tests.

**Comment by basil** [comment]
The original author of Timestamper used EqualsVerifier extensively. But I think if we adopted it in Jenkins core, the cost of rebuilding failed Dependabot PRs would be greater than the benefit we'd get from using this library in a single test.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
+1, no need to keep the lib just for one test

**Pull request merged**
